atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/ValueStoreImpl.java

```diff
@@ -136,11 +141,18 @@ private Stream<CellReference> extractCandidateCells(LockDescriptor descriptor) {
     }
 
     private void applyLockedDescriptors(Set<LockDescriptor> lockDescriptors) {
-        lockDescriptors.stream().flatMap(this::extractCandidateCells).forEach(this::putLockedCell);
+        lockDescriptors.stream()
+                // Explicitly exclude descriptors corresponding to watched rows from non-watched tables
+                .filter(this::isTableWatched)
```

Comment:
todo - double-check if this is needed

Reply:
Yup, it is. otherwise the snapshot will contain locked row references, which (if I remember correctly from 2 weeks ago) is not the desired behaviour - confirmed by unit tests.

Reply:
regrettable that we need to parse this twice - once to get the table ref, then immediately again to get candidate cells. I wonder if we can optimise this so that we don't waste that parsing time

Reply:
yep, done with a small refactor :) 

I'm not convinced by the name `TableRefAndRemainder` - would `ParsedLockDescriptor` be more suitable?